2016-02-03  Omar B Ozgur  <ozgur@lnxsrv09.seas.ucla.edu>

        * timespec.h: timespec_cmp now returns -1, 0, or 1 if A < B, A == B,
        or A > B repectively.

--- coreutils-with-bug/lib/timespec.h	2005-09-21 23:05:39.000000000 -0700
+++ coreutils-with-bug-changed/lib/timespec.h	2016-02-03 18:08:57.145108000 -0800
@@ -45,8 +45,11 @@
 static inline int
 timespec_cmp (struct timespec a, struct timespec b)
 {
-  int diff = a.tv_sec - b.tv_sec;
-  return diff ? diff : a.tv_nsec - b.tv_nsec;
+  if(a.tv_sec > b.tv_sec)
+	return 1;
+  else if(a.tv_sec < b.tv_sec)
+	return -1;
+  return 0;
 }
 
 # if ! HAVE_DECL_NANOSLEEP
